### DISCUSSION TITLE
FROLL and Homematic, Channel 3 or 4?

### DISCUSSION THREAD

**Discussion by Pinky007** [OpenCCU/OpenCCU#1409] [🇩🇪 Fragen&Antworten] [answered]
Hello,
I was looking on getting the value of my FROLL in Home Assistant but it is only displaying 0 or 1 (means completely open or closed) even if I stop it manually.
There is a quite good description of the problem here:
[https://homematic-forum.de/forum/viewtopic.php?f=65&t=66116](https://homematic-forum.de/forum/viewtopic.php?f=65&t=66116)

It seems that Home Assistant is getting the channel 4 for the sensor.shutter_xxxxx_level while it should be getting channel 3.
Is that done on purpose? Is there an other way that I missed to get the position value? Everything is working well in Homematic (3.59.6.20210911) and I can get the right value there but it is not well connected to Home assistant.

Let me know if I can support if further testing if this would be needed.
Thank you!

**Comment by Pinky007** [comment]
Hallo @jens-maus ,
Hast du das Problem schon gesehen?
Vielen Dank!

**Reply by jens-maus** [reply]
Nope, didn't notice it, but also don't have a working HmIP-FROLL connected here on my HA test environment.

**Reply by Pinky007** [reply]
Would it be possible to exchange the channels in the data that you send to HA? (to make a test at least)

**Comment by Baxxy13** [comment]
Die einzelnen Datenpunkte/Parameter sind festgelegt und werden so an externe Logiken (HA / ioBroker) übergeben. Da kann RaspberryMatic nichts dran ändern. Was die externen Logiken daraus machen liegt auf deren Seite. 
Das "Problem" mit deinem FROLL besteht ja darin das Kanal:3 zwar die reale Behanghöhe ausgibt, der FROLL aber über die Kanäle:4/5/6 und nicht über Kanal:3 gesteuert wird. In der Homeassistant-Community taucht das Problem ab und zu auf, hier wäre vielleicht ein Lösungsansatz für Dich. 
[Possibility to show current shutter-position?](https://community.home-assistant.io/t/possibility-to-show-current-shutter-position/207084)

Ich teste auch nebenbei die HA-Implementierung von HM und HmIP-Komponenten. Muss aber sagen das es gerade für IP teilweise echt unausgereift wirkt. Einfaches Beispiel: Eine HmIP-PS, bei der ich nur Kanal:3 schalten kann. Für Kanal:4/5 muss ich erstmal umständlich irgendwelche Konfigurationen anlegen.

**Reply by Pinky007** [reply]
Hi,
Vielen Dan für die Antwort und das Link.
Leider hilft das nicht wirklich da die Daten gar nicht für das FROLL geliefert worden.

Das Template der in dein Link erwähnt ist liefert:
`<template TemplateState(<state cover.shutter_kitchen_sink=closed; current_position=0, id=00115A49A5B164, interface=hmip, rssi_peer=False, level=False, friendly_name=Shutter Kitchen Sink, icon=hass:window-shutter, supported_features=15 @ 2021-10-03T15:41:25.889250+02:00>)>`
wenn es runtergefahren (0) worden aber nicht ganz nach unten
und 
`<template TemplateState(<state cover.shutter_kitchen_sink=open; current_position=100, id=00115A49A5B164, interface=hmip, rssi_peer=-58, level=1.0, friendly_name=Shutter Kitchen Sink, icon=hass:window-shutter, supported_features=15 @ 2021-10-03T15:42:32.568612+02:00>)>`
wenn es hoch gefahren (100) worden aber nicht ganz nach oben

Mann kann es grundsätzlich auch in die state attributes sehen:
![grafik](https://user-images.githubusercontent.com/1488257/135756300-055c681e-86f6-4ca3-9257-a81bd8e39ca8.png)

Leider denke ich dass hier das Problem kommt von das Mapping zwischen die Channel zwischen HM und hass.
Besteht ein möglichkeit diese mapping zu anpassen?

**Reply by jens-maus** [reply]
Da für die Interaktion zwischen HomeAssistant und RaspberryMatic/CCU die pyhomematic (https://github.com/danielperna84/pyhomematic) Bibliothek zuständig ist müsste vermutlich die Frage dort gestellt werden bzw @danielperna84 darüber Auskunft geben.

**Comment by danielperna84** [comment, marked as answer]
Das Thema ist ein Dauerbrenner und wurde schon zigfach erläutert an lauter unterschiedlichen stellen.

Um es zusammenzufassen:  
Die _alten_ HM-Geräte haben für Ist-Zustand und Soll-Zustand den gleichen Kanal verwendet. Man konnte also z.B. `LEVEL` auf Kanal 1 schreiben um den Gerätezustand zu ändern, und wenn man es gelesen hat, dann hat man den tatsächlichen Wert bekommen. Auf dieser Basis wurde pyhomematic entwickelt, und funktioniert auch nur mit diesem Konzept sinnvoll.

Bei HomeMatic IP wurde entschieden das alles auf 4 Kanäle aufzuteilen. Der erste hat den tatsächlichen Status, die anderen drei sind die virtuellen Kanäle (wovon wir in Home Assistant immer nur den ersten benutzen - wer auf der CCU die Expertenansicht deaktiviert hat würde sonst nicht verstehen weshalb er 3 Aktoren für nur ein physisches Gerät hat). Das lässt sich nachträglich nicht so ohne weiteres in pyhomematic (und somit die Home Assistant Integration) einbauen ohne komplett das Rad neu zu erfinden.

Meine Idee war es nun die alte, auf pyhomematic basierende Integration auslaufen zu lassen und _sauber_ (ohne Altlasten) mit [hahomematic](https://github.com/danielperna84/hahomematic) neu anzufangen. Wäre vom Konzept her anders, und würde solche Fälle leichter implementierbar machen. Das Problem: es scheint in der HomeMatic community keine Python-Entwickler zu geben die Zeit und / oder Lust haben zu helfen. Und die die's kümmert sind nur zum testen qualifiziert, was zum aktuellen Stand aber noch nicht nötig ist. Ergo liegt das Projekt bei mir derzeit auf Eis und es gibt in absehbarer Zeit nicht viel neues. Es gibt [hier](https://github.com/home-assistant/core/pull/56619) einen pull request, der bei dem Problem helfen würde. Das ist aber auch nur ein Workaround, und bisher noch nicht in Home Assistant integriert.

**Comment by Guenter-1953** [comment]
Es geht um den HMIP-FROLL. Sollte ich hier falsch sein, dann gebt mir bitte einen Hinweis auf Hilfe. Danke vorweg.

Ich habe 124 Geräte an der Raspmatic und bin sehr zufrieden. Tolle Konstellation. Nur ein HMIP-FROLL von 10 macht Probleme. Ich habe dies an ELV geschrieben aber die verweigern mir die Hilfe mit dem lapidaren Satz: 

> In Ihrem Fall ist eine RaspberryMatic-Zentrale und keine CCU3 Zentrale > gegeben.
> Einen Support für die Verhaltensweise von Geräten an Communitylösungen, wie 
> z.B. RaspberryMatic, DepMatic o.ä., bieten wir im Rahmen unseres 
> Kundenservice nicht an. Daher können wir Ihnen zu Ihrer Anfrage keine 
> Unterstützung anbieten.

Ich habe direkt bei ELV einen 3er Pack gekauft und alle anderen gehen. Hier das Problem das ich so auch an ELV geschrieben habe:

> > Sehr geehrte Damen und Herren,
> >
> > wir haben einige HmIP-FROLL im Einsatz. Einer verhält sich jedoch nicht so
> > wie er sollte. Bei, Drücken des Taster "AUF" bewegt er sich auch auf, aber
> > wenn man ihn zwischendurch anhalten will, so muss man die "AB" drücken. Bei
> > einer Abwärtsfahrt genau das umgekehrte Verhalten.
> >
> > Alle unsere Anderen HmIP-FROLL funktionieren einwandfrei. Also bei "AUF"
> > stoppen mit AUF und bei "AB" stoppen bei AB.
> >
> > Dieser HmIP-FROLL ist identisch konfiguriert wie alle anderen HmIP-FROLL.
> > Er hat die Seriennummer 00115D899C79B7  mit der Firmware 1.8.12
> >
> > Die Einstellung Schnellere Reaktion bei erneutem Tastendruck durch Abbruch
> > alter Sendevorgänge haben wir in allen möglichen Kombinationen ausprobiert,
> > es geht nicht. Wie geschrieben alle Hm-IP FROLL sind gleich eingestellt.
> >
> > Was ist zu tuen??

Danke für Eure Antwort
